docs/01-app/04-api-reference/04-functions/use-link-status.mdx

```diff
@@ -0,0 +1,76 @@
+---
+title: useLinkStatus
+description: API Reference for the useLinkStatus hook.
+---
+
+`useLinkStatus` is a **Client Component** hook that lets you track the loading state of a `Link` component during navigation. It's particularly useful when you want to show loading indicators during page transitions, especially when prefetching is disabled.
+
+```tsx filename="app/example-client-component.tsx" switcher
+'use client'
+
+import { useLinkStatus } from 'next/navigation'
+import Link from 'next/link'
+
+function LoadingIndicator() {
+  const { pending } = useLinkStatus()
+  return pending ? <div>Loading...</div> : null
+}
+
+export default function ExampleClientComponent() {
+  return (
+    <Link href="/dashboard" prefetch={false}>
+      Dashboard <LoadingIndicator />
+    </Link>
```

Comment:
Since `next/link` is a client component, would you already be in the client world? 

It'll be good to use a real-world example here to help people understand how it's used. Let's say we have a `<Header/>` server component, and `<LoadingIndicator>` is a client component, can we nest them? Or do we need `use client` since `<Link>` is a client component?